src/main/java/com/example/homework6/controllers/EmailController.java

```diff
@@ -0,0 +1,156 @@
+package com.example.homework6.controllers;
+
+import com.example.homework6.*;
+import com.example.homework6.repos.EmailRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.Optional;
+
+@Controller
+public class EmailController {
+
+    @Autowired
+    EmailRepo emailRepo;
+
+    @Autowired
+    EmailSender emailSender;
+
+    @Autowired
+    TaskSchedule taskSchedule;
+
+
+    @GetMapping(value = "/create")
+    private String createEmail(Model model) {
+        model.addAttribute("emailItem", new EmailItem());
+        return "create";
+    }
+
+    @PostMapping(value = "/saveEmail")
+    private String saveEmail(@Valid @ModelAttribute("emailItem") EmailItem emailItem, BindingResult result){
+
+        if(result.hasErrors()){
+            return "create";
+        }else if(!DateValidator.validate(emailItem.getDeliveryDate())){
+            FieldError error = new FieldError("emailItem", "deliveryDate",
+                    "Date is wrong! Please, be carefully! DATE format is 'dd-MM-yyyy HH:mm'!");
+            result.addError(error);
+            return "create";
+        }
+
+        emailRepo.save(emailItem);
+        return "index";
+    }
+
+    @GetMapping(value = "/emails")
+    private ModelAndView getEmails(){
+        Iterable<EmailItem> emails = emailRepo.findAll();
+        ModelAndView mv = new ModelAndView();
+        mv.setViewName("emails");
+        mv.addObject("emails", emails);
+        mv.getModel();
+        return mv;
+    }
+
+    @GetMapping(value = "/email")
```

Comment:
Make this url mapping more specific. URL is the name. If you want to get specific email you need to provide unique name to identify it from the others. Consider, adding id as url path part

Reply:
I'm using this url for call modal window for editing email.